static/widgets/libs-widget.ts

```diff
@@ -41,6 +41,76 @@ type FavLibraries = Record<string, string[]>;
 
 type PopupAlertFilter = (compiler: string, langId: string) => {title: string; content: string} | null;
 
+type LibraryAnnotation = {
+    commithash?: string;
+    cxx11?: boolean;
+    machine?: string;
+    osabi?: string;
+    error?: string;
+};
+
+type LibraryBuildInfo = {
+    arch?: string;
+    compilerId?: string;
+    compilerType?: string;
+    libcxx?: string;
+    os?: string;
+    error?: string;
+};
+
+type LibraryAnnotationDetail = {
+    buildhash: string;
+    annotation: LibraryAnnotation;
+    buildinfo: LibraryBuildInfo;
+};
+
+class LibraryAnnotations {
+    private all: Record<string, LibraryAnnotationDetail[]> = {};
+    constructor() {}
+
+    private async get(library: string, version: string): Promise<LibraryAnnotationDetail[]> {
+        const libver = `${library}/${version}`;
+        if (!Object.keys(this.all).includes(libver)) {
+            const response = await fetch(`https://conan.compiler-explorer.com/annotations/${libver}`);
```

Comment:
Is this the first time we'll be hitting `conan.compiler-explorer.com` externally? Presuming cloudfront caching is enough? or do we need to stand up somethign a bit more substantial to handle this? (we could cache ourselves in a lambda if it became an issue?) (obviously this caches here locally too)

Reply:
First time indeed of frontend knowing about conan.compiler-explorer.com, also cloud-front cache is currently disabled, and thats ok in this case. People tend to not ctrl+r refresh until really needed, and that might restrain the IO, though the request itself or library and version is quite intensive. I think we'll have to monitor the situation.